{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { SyntaxError: Unexpected string in JSON at position 324
          at JSON.parse (<anonymous>)
          at parse (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\body-parser\lib\types\json.js:89:19)
          at C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\body-parser\lib\read.js:121:18
          at invokeCallback (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:224:16)
          at done (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:213:7)
          at IncomingMessage.onEnd (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:273:7)
          at IncomingMessage.emit (events.js:203:15)
          at IncomingMessage.EventEmitter.emit (domain.js:448:20)
          at endReadableNT (_stream_readable.js:1145:12)
          at process._tickCallback (internal/process/next_tick.js:63:19)
        expose: true,
        statusCode: 400,
        status: 400,
        body:
         '\r\n    {\r\n        "firstName": "Brahim",\r\n        "lastName": "Afassy",\r\n        "email": "s@s.com",\r\n        "CNE": "JC563288",\r\n        "tele": "088763652",\r\n        "gender": "Male",\r\n        "CNI": "D1325152",\r\n        "birthdate": "1999-02-14T00:00:00.000Z",\r\n        "group": {\r\n            "code": "ISIL"\r\n            "name":"I...S...I...L..."\r\n        },\r\n        "password": "123ABCabc",\r\n        "__v": 0,\r\n        "absenceAVG": 10\r\n    }',
        type: 'entity.parse.failed' } },
  level: 'error',
  timestamp: '2020-01-09 12:27:20' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { MongoError: E11000 duplicate key error collection: EU.students index: email_1 dup key: { email: "s@s.com" }
          at Function.create (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongodb\lib\core\error.js:44:12)
          at toError (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongodb\lib\utils.js:150:22)
          at coll.s.topology.insert (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongodb\lib\operations\common_functions.js:266:39)
          at handler (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongodb\lib\core\sdam\topology.js:971:24)
          at wireProtocol.(anonymous function) (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongodb\lib\core\sdam\server.js:496:5)
          at C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongodb\lib\core\connection\pool.js:414:18
          at process._tickCallback (internal/process/next_tick.js:61:11)
        driver: true,
        name: 'MongoError',
        index: 0,
        code: 11000,
        keyPattern: { email: 1 },
        keyValue: { email: 's@s.com' },
        errmsg:
         'E11000 duplicate key error collection: EU.students index: email_1 dup key: { email: "s@s.com" }',
        [Symbol(mongoErrorContextSymbol)]: {} } },
  level: 'error',
  timestamp: '2020-01-09 12:28:47' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { SyntaxError: Unexpected string in JSON at position 36
          at JSON.parse (<anonymous>)
          at parse (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\body-parser\lib\types\json.js:89:19)
          at C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\body-parser\lib\read.js:121:18
          at invokeCallback (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:224:16)
          at done (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:213:7)
          at IncomingMessage.onEnd (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:273:7)
          at IncomingMessage.emit (events.js:203:15)
          at IncomingMessage.EventEmitter.emit (domain.js:448:20)
          at endReadableNT (_stream_readable.js:1145:12)
          at process._tickCallback (internal/process/next_tick.js:63:19)
        expose: true,
        statusCode: 400,
        status: 400,
        body:
         '\r\n    {\r\n    \t"role" : "s"\r\n        "firstName": "Brahim",\r\n        "lastName": "Afassy",\r\n        "email": "c@c.com",\r\n        "CNE": "JC563288",\r\n        "tele": "088763652",\r\n        "gender": "Male",\r\n        "CNI": "D1325152",\r\n        "birthdate": "1999-02-14T00:00:00.000Z",\r\n        "group": {\r\n            "code": "ISIL",\r\n            "name":"I...S...I...L..."\r\n        },\r\n        "password": "123ABCabc",\r\n        "absenceAVG": 10\r\n    }',
        type: 'entity.parse.failed' } },
  level: 'error',
  timestamp: '2020-01-11 04:06:19' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      ReferenceError: prof is not defined
          at _callee2$ (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\src\routes/User.js:18:38)
          at tryCatch (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:45:40)
          at Generator.invoke [as _invoke] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:271:22)
          at Generator.prototype.(anonymous function) [as next] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:97:21)
          at tryCatch (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:45:40)
          at invoke (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:135:20)
          at C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:145:13
          at process._tickCallback (internal/process/next_tick.js:68:7) },
  level: 'error',
  timestamp: '2020-01-11 04:27:35' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { SyntaxError: Unexpected token { in JSON at position 3
          at JSON.parse (<anonymous>)
          at parse (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\body-parser\lib\types\json.js:89:19)
          at C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\body-parser\lib\read.js:121:18
          at invokeCallback (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:224:16)
          at done (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:213:7)
          at IncomingMessage.onEnd (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:273:7)
          at IncomingMessage.emit (events.js:203:15)
          at IncomingMessage.EventEmitter.emit (domain.js:448:20)
          at endReadableNT (_stream_readable.js:1145:12)
          at process._tickCallback (internal/process/next_tick.js:63:19)
        expose: true,
        statusCode: 400,
        status: 400,
        body:
         '{\n\t{\n        "pictur": "http://www.ietpune.com/img/teacher-img/user.png",\n        "absenceAVG": 10,\n        "_id": "5e193bdf1aa00d3da0916a19",\n        "role": "Professor",\n        "firstName": "Brahim",\n        "lastName": "Afassy",\n        "email": "b@b.com",\n        "CNE": "JC563288",\n        "tele": "088763652",\n        "gender": "Male",\n        "CNI": "D1325152",\n        "birthdate": "1999-02-14T00:00:00.000Z",\n        "group": [\n            {\n                "_id": "5e193bdf1aa00d3da0916a1a",\n                "code": "ISIL",\n                "name": "I...S...I...L..."\n            }\n        ],\n        "password": "123ABCabc",\n    }\n}',
        type: 'entity.parse.failed' } },
  level: 'error',
  timestamp: '2020-01-11 07:12:02' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { SyntaxError: Unexpected token { in JSON at position 3
          at JSON.parse (<anonymous>)
          at parse (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\body-parser\lib\types\json.js:89:19)
          at C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\body-parser\lib\read.js:121:18
          at invokeCallback (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:224:16)
          at done (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:213:7)
          at IncomingMessage.onEnd (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\raw-body\index.js:273:7)
          at IncomingMessage.emit (events.js:203:15)
          at IncomingMessage.EventEmitter.emit (domain.js:448:20)
          at endReadableNT (_stream_readable.js:1145:12)
          at process._tickCallback (internal/process/next_tick.js:63:19)
        expose: true,
        statusCode: 400,
        status: 400,
        body:
         '{\n\t{\n        "pictur": "http://www.ietpune.com/img/teacher-img/user.png",\n        "absenceAVG": 10,\n        "_id": "5e193bdf1aa00d3da0916a19",\n        "role": "Professor",\n        "firstName": "Brahim",\n        "lastName": "Afassy",\n        "email": "b@b.com",\n        "CNE": "JC563288",\n        "tele": "088763652",\n        "gender": "Male",\n        "CNI": "D1325152",\n        "birthdate": "1999-02-14T00:00:00.000Z",\n        "group": [\n            {\n                "code": "ISIL",\n                "name": "I...S...I...L..."\n            }\n        ],\n        "password": "123ABCabc"\n    }\n}',
        type: 'entity.parse.failed' } },
  level: 'error',
  timestamp: '2020-01-11 07:12:16' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { CastError: Cast to ObjectId failed for value "undefined" at path "_id" for model "User"
          at new CastError (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\error\cast.js:29:11)
          at ObjectId.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schema\objectid.js:246:11)
          at ObjectId.SchemaType.applySetters (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:969:12)
          at ObjectId.SchemaType._castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1383:15)
          at ObjectId.SchemaType.castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1373:15)
          at ObjectId.SchemaType.castForQueryWrapper (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1352:15)
          at cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\cast.js:315:32)
          at model.Query.Query.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4641:12)
          at castQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4454:18)
          at model.Query.Query._findAndModify (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3417:23)
          at model.Query.<anonymous> (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3380:8)
          at model.Query._wrappedThunk [as _findOneAndRemove] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\helpers\query\wrapThunk.js:16:8)
          at process.nextTick (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\kareem\index.js:369:33)
          at process._tickCallback (internal/process/next_tick.js:61:11)
        message:
         'Cast to ObjectId failed for value "undefined" at path "_id" for model "User"',
        name: 'CastError',
        stringValue: '"undefined"',
        kind: 'ObjectId',
        value: 'undefined',
        path: '_id',
        reason: undefined,
        model: Model { User } } },
  level: 'error',
  timestamp: '2020-01-11 07:38:26' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { CastError: Cast to ObjectId failed for value "undefined" at path "_id" for model "User"
          at new CastError (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\error\cast.js:29:11)
          at ObjectId.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schema\objectid.js:246:11)
          at ObjectId.SchemaType.applySetters (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:969:12)
          at ObjectId.SchemaType._castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1383:15)
          at ObjectId.SchemaType.castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1373:15)
          at ObjectId.SchemaType.castForQueryWrapper (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1352:15)
          at cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\cast.js:315:32)
          at model.Query.Query.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4641:12)
          at castQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4454:18)
          at model.Query.Query._findAndModify (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3417:23)
          at model.Query.<anonymous> (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3380:8)
          at model.Query._wrappedThunk [as _findOneAndRemove] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\helpers\query\wrapThunk.js:16:8)
          at process.nextTick (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\kareem\index.js:369:33)
          at process._tickCallback (internal/process/next_tick.js:61:11)
        message:
         'Cast to ObjectId failed for value "undefined" at path "_id" for model "User"',
        name: 'CastError',
        stringValue: '"undefined"',
        kind: 'ObjectId',
        value: 'undefined',
        path: '_id',
        reason: undefined,
        model: Model { User } } },
  level: 'error',
  timestamp: '2020-01-11 07:40:07' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { CastError: Cast to ObjectId failed for value "undefined" at path "_id" for model "User"
          at new CastError (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\error\cast.js:29:11)
          at ObjectId.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schema\objectid.js:246:11)
          at ObjectId.SchemaType.applySetters (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:969:12)
          at ObjectId.SchemaType._castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1383:15)
          at ObjectId.SchemaType.castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1373:15)
          at ObjectId.SchemaType.castForQueryWrapper (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1352:15)
          at cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\cast.js:315:32)
          at model.Query.Query.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4641:12)
          at castQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4454:18)
          at model.Query.Query._findAndModify (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3417:23)
          at model.Query.<anonymous> (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3380:8)
          at model.Query._wrappedThunk [as _findOneAndRemove] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\helpers\query\wrapThunk.js:16:8)
          at process.nextTick (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\kareem\index.js:369:33)
          at process._tickCallback (internal/process/next_tick.js:61:11)
        message:
         'Cast to ObjectId failed for value "undefined" at path "_id" for model "User"',
        name: 'CastError',
        stringValue: '"undefined"',
        kind: 'ObjectId',
        value: 'undefined',
        path: '_id',
        reason: undefined,
        model: Model { User } } },
  level: 'error',
  timestamp: '2020-01-11 07:42:42' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { CastError: Cast to ObjectId failed for value "undefined" at path "_id" for model "User"
          at new CastError (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\error\cast.js:29:11)
          at ObjectId.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schema\objectid.js:246:11)
          at ObjectId.SchemaType.applySetters (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:969:12)
          at ObjectId.SchemaType._castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1383:15)
          at ObjectId.SchemaType.castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1373:15)
          at ObjectId.SchemaType.castForQueryWrapper (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1352:15)
          at cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\cast.js:315:32)
          at model.Query.Query.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4641:12)
          at castQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4454:18)
          at model.Query.Query._findAndModify (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3417:23)
          at model.Query.<anonymous> (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3380:8)
          at model.Query._wrappedThunk [as _findOneAndRemove] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\helpers\query\wrapThunk.js:16:8)
          at process.nextTick (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\kareem\index.js:369:33)
          at process._tickCallback (internal/process/next_tick.js:61:11)
        message:
         'Cast to ObjectId failed for value "undefined" at path "_id" for model "User"',
        name: 'CastError',
        stringValue: '"undefined"',
        kind: 'ObjectId',
        value: 'undefined',
        path: '_id',
        reason: undefined,
        model: Model { User } } },
  level: 'error',
  timestamp: '2020-01-11 07:43:33' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { CastError: Cast to ObjectId failed for value "undefined" at path "_id" for model "User"
          at new CastError (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\error\cast.js:29:11)
          at ObjectId.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schema\objectid.js:246:11)
          at ObjectId.SchemaType.applySetters (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:969:12)
          at ObjectId.SchemaType._castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1383:15)
          at ObjectId.SchemaType.castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1373:15)
          at ObjectId.SchemaType.castForQueryWrapper (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1352:15)
          at cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\cast.js:315:32)
          at model.Query.Query.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4641:12)
          at castQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4454:18)
          at model.Query.Query._findAndModify (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3417:23)
          at model.Query.<anonymous> (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3380:8)
          at model.Query._wrappedThunk [as _findOneAndRemove] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\helpers\query\wrapThunk.js:16:8)
          at process.nextTick (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\kareem\index.js:369:33)
          at process._tickCallback (internal/process/next_tick.js:61:11)
        message:
         'Cast to ObjectId failed for value "undefined" at path "_id" for model "User"',
        name: 'CastError',
        stringValue: '"undefined"',
        kind: 'ObjectId',
        value: 'undefined',
        path: '_id',
        reason: undefined,
        model: Model { User } } },
  level: 'error',
  timestamp: '2020-01-11 07:43:58' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      { CastError: Cast to ObjectId failed for value "undefined" at path "_id" for model "User"
          at new CastError (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\error\cast.js:29:11)
          at ObjectId.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schema\objectid.js:246:11)
          at ObjectId.SchemaType.applySetters (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:969:12)
          at ObjectId.SchemaType._castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1383:15)
          at ObjectId.SchemaType.castForQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1373:15)
          at ObjectId.SchemaType.castForQueryWrapper (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\schematype.js:1352:15)
          at cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\cast.js:315:32)
          at model.Query.Query.cast (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4641:12)
          at castQuery (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:4454:18)
          at model.Query.Query._findAndModify (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3417:23)
          at model.Query.<anonymous> (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\query.js:3380:8)
          at model.Query._wrappedThunk [as _findOneAndRemove] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\mongoose\lib\helpers\query\wrapThunk.js:16:8)
          at process.nextTick (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\kareem\index.js:369:33)
          at process._tickCallback (internal/process/next_tick.js:61:11)
        message:
         'Cast to ObjectId failed for value "undefined" at path "_id" for model "User"',
        name: 'CastError',
        stringValue: '"undefined"',
        kind: 'ObjectId',
        value: 'undefined',
        path: '_id',
        reason: undefined,
        model: Model { User } } },
  level: 'error',
  timestamp: '2020-01-11 07:45:07' }
{ message:
   { Code: 'MiddlewareErrorHandler',
     error:
      ReferenceError: ComparePassword is not defined
          at _callee5$ (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\src\routes/User.js:61:47)
          at tryCatch (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:45:40)
          at Generator.invoke [as _invoke] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:271:22)
          at Generator.prototype.(anonymous function) [as next] (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:97:21)
          at tryCatch (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:45:40)
          at invoke (C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:135:20)
          at C:\Users\brahi\Desktop\MERN\MERN_Project\Server\node_modules\regenerator-runtime\runtime.js:145:13
          at process._tickCallback (internal/process/next_tick.js:68:7) },
  level: 'error',
  timestamp: '2020-01-11 22:07:13' }
{ message:
   { Code: 'uncaughtException',
     error:
      { Error: ENOENT: no such file or directory, open 'C:\Users\brahi\Desktop\MERN\MERN_Project\server\src\Config\logs\morgan-logs.log'
        errno: -4058,
        code: 'ENOENT',
        syscall: 'open',
        path:
         'C:\\Users\\brahi\\Desktop\\MERN\\MERN_Project\\server\\src\\Config\\logs\\morgan-logs.log' } },
  level: 'error',
  timestamp: '2020-01-11 23:10:39' }
{ message:
   { Code: 'uncaughtException',
     error:
      { Error: ENOENT: no such file or directory, open 'C:\Users\brahi\Desktop\MERN\MERN_Project\server\src\Config\logs\morgan-logs.log'
        errno: -4058,
        code: 'ENOENT',
        syscall: 'open',
        path:
         'C:\\Users\\brahi\\Desktop\\MERN\\MERN_Project\\server\\src\\Config\\logs\\morgan-logs.log' } },
  level: 'error',
  timestamp: '2020-01-11 23:11:54' }
